doc/userguide/output/eve/eve-json-format.rst

```diff
@@ -1287,3 +1287,315 @@ Example of RFB logging, with full VNC style authentication parameters:
         "blue_shift": 0
       }
     }
+
+
+Event type: MQTT
+----------------
+
+EVE-JSON output for MQTT consists of one object per MQTT message, with some common and various type-specific fields. Note that some message types (aka control packet types), such as ``PINGREQ`` and ``PINGRESP``, have no type-specific data. These only contain the common fields listed below.
+
+
+Common fields
+~~~~~~~~~~~~~
+
+Common fields from the MQTT fixed header:
+
+* "message_type": MQTT message type as string, e.g. ``CONNECT``, ``PUBLISH``, ...
+* "qos": Quality of service level for the message, integer between 0 and 2.
+* "retain": Boolean value of the MQTT 'retain' flag.
+* "dup": Boolean value of the MQTT 'dup' (duplicate) flag.
+
+
+MQTT CONNECT fields
+~~~~~~~~~~~~~~~~~~~
+
+* "connect.protocol_string": Protocol string as defined in the spec, e.g. ``MQTT`` (MQTT 3.1.1 and later) or ``MQIsdp`` (MQTT 3.1).
+*  "protocol_version": Protocol version as defined in the spec:
+
+   * protocol version ``3``: MQTT 3.1
+   * protocol version ``4``: MQTT 3.1.1
+   * protocol version ``5``: MQTT 5.0
+
+* "connect.flags.username", "connect.flags.password":  Set to `true` if credentials are submitted with the connect request.
+* "connect.flags.will": Set to `true` if a will is set.
+* "connect.flags.will_retain": Set to `true` if the will is to be retained on the broker.
+* "connect.will.clean_session": Set to `true` if the connection is to made with a clean session.
+* "connect.client_id": Client ID string submitted my the connecting client.
+* "connect.username", "connect.password":  User/password authentication credentials submitted with the connect request.
+* "connect.will.topic": Topic to publish the will message to.
+* "connect.will.message": Message to be published on connection loss.
+* "connect.will.properties": (Optional, MQTT 5.0) Will properties set on this request. See `3.1.3.2 in the spec <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901060>`_ for more information on will properties.
+* "connect.properties": (Optional, MQTT 5.0) CONNECT properties set on this request. See `3.1.2.11 in the spec <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901046>`_ for more information on CONNECT properties.
+
+Example of MQTT CONNECT logging:
+
+::
+
+  "mqtt": {
+    "message_type": "CONNECT",
+    "qos": 0,
+    "retain": false,
+    "dup": false,
+    "connect": {
+      "protocol_string": "MQTT",
+      "protocol_version": 5,
+      "flags": {
+        "username": true,
+        "password": true,
+        "will_retain": false,
+        "will": true,
+        "clean_session": true
+      },
+      "client_id": "client",
+      "username": "user",
+      "password": "pass",
```

Comment:
I think that is doable. Thanks for the idea.

Reply:
Done. Will be in next PR.

Reply:
FYI: This will only be configurable in the first iteration for `event_type: mqtt` log output. There is one small exception: I have also implemented MQTT metadata output in `alert` events, but since I could not find a way to access the configuration items of another type in the same EVE output, I had to settle for a default, which is to log passwords. 
I took some inspiration from the DNS code, which also falls back to a default setting of "output all rrtypes" in alerts, regardless of what is configured in the `dns` type.